cluster-autoscaler/cloudprovider/clusterapi/clusterapi_controller.go

```diff
@@ -203,34 +235,56 @@ func (c *machineController) findScalableResourceByProviderID(providerID normaliz
 		return nil, err
 	}
 
-	if machine == nil {
-		return nil, nil
-	}
-
```

Comment:
I removed this check because `machine` is only used to look up the MachineSet. Now that we support MachinePools, it looks possible to me that there's a path through this function `findScalableResourceByProviderID` where `machine` and/or `machineSet` are `nil` but we still find `machinePool` successfully via its providerID.

It could be considered a logical error still if `machine == nil` (there *should* be a machine, right?), but in terms of the code it's not relevant to looking up the MachinePool. But maybe we should reinstate the `if machine == nil` check? I'm not sure.

Reply:
would it make sense to add some sort of conditional logic that would allow a nil machine/machineset to process the machinepool section?

Reply:
We could test for MachinePool first, then execute the rest of the function basically as-is. (That's a bit cleaner IMHO, but I changed this because we thought MachinePools should be checked last since they're probably less common that MachineDeployments/MachineSets.)

Reply:
@elmiko I think it's safer to revert to the way it was previously, although arguably checking for MP first is less intuitive. I added a commit with that change, LMK whatcha think.

Reply:
+1 @mboersma , sorry for the flapping here. i think putting it first does make more sense, my only ask is to add a comment here describing why we put it first so we don't forget later.